packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote/interceptors/auth_interceptor.dart

```diff
@@ -49,24 +52,13 @@ class AuthInterceptor extends Interceptor {
   @override
   Future<void> onError(DioError err, ErrorInterceptorHandler handler) async {
     if (err.response?.statusCode == 401) {
-      final newToken = await loadNewToken();
+      final newToken = await _fetchAuthTokenUseCase.fetchNewAccessToken();
       if (newToken == null) {
         await _logoutUseCase.execute();
       }
     }
     super.onError(err, handler);
   }
 
-  Future<String?> loadNewToken() async {
-    final refreshToken = await getRefreshToken();
-    if (refreshToken == null) {
-      return null;
-    }
-    final newToken = await _authRepository.fetchNewToken();
-    return newToken;
-  }
-
   Future<String?> getToken() async => _authRepository.getToken();
```

Comment:
Could we also move this as part of the `FetchAuthTokenUseCase` (although, name it accordingly)? This way we'd remove one dependency from `AuthInterceptor`

Reply:
Regarding the comment from Stanev "Each UseCase have to have just one public method, and should be named 'execute'", is it a good idea to move it in separate UseCase or it is better to leave it here? What is your suggestion?

Reply:
What I asked was if it would make sense to move the `getToken()` to be part of the `FetchAuthTokenUseCase`. The current code in the FetchAuthTokenUseCase is as follows:
```
final refreshToken = await _authRepository.getRefreshToken(); 
if (refreshToken == null) { 
     return null;
}
final newToken = await _authRepository.fetchNewToken();
return newToken;
```
Would it be okay if we return the token with `getToken()` if it's not null, or if it is execute the rest of the code above?